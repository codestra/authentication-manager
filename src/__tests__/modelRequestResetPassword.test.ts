import { MongoMemoryServer } from 'mongodb-memory-server';
import mongoose from 'mongoose';

process.env = { PASSWORD_RESET_EXPIRES: '8640000', JWT_EXPIRES: '9999y', JWT_SECRET: '123123' };
import { modelSignUp, modelRequestResetPassword } from '../index';
import MockModel from '../mocks/model';

describe('modelSignUp', () => {
  let mongoServer: MongoMemoryServer;

  beforeAll(async () => {
    mongoServer = await MongoMemoryServer.create();
    await mongoose.connect(mongoServer.getUri(), { useNewUrlParser: true, useUnifiedTopology: true });
  });

  afterAll(async () => {
    if (mongoServer) {
      await mongoServer.stop();
      await mongoose.disconnect();
    }
  });

  it('throw an error when no email has been found', async () => {
    // wrong token
    await expect(async () => {
      await modelRequestResetPassword({ Model: MockModel, variables: { email: 'sajad.ghawami@codestra.io' } });
    }).rejects.toThrowErrorMatchingSnapshot();
  });

  it('signed up model and reset the password twice to see if they differ', async () => {
    await modelSignUp({
      Model: MockModel,
      variables: { email: 'sajad.ghawami@codestra.io', password: '123123' },
    });

    const onCompleted = jest.fn();
    const passwordResetToken1 = await modelRequestResetPassword({
      Model: MockModel,
      variables: { email: 'sajad.ghawami@codestra.io' },
      onCompleted,
    });
    const passwordResetToken2 = await modelRequestResetPassword({
      Model: MockModel,
      variables: { email: 'sajad.ghawami@codestra.io' },
    });

    expect(passwordResetToken1).not.toBe(passwordResetToken2);
    expect(onCompleted).toHaveBeenCalled();
  });
});
