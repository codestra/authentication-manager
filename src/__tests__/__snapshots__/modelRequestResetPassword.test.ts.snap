// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modelSignUp throw an error when no email has been found 1`] = `"Error: ERROR_USER_NOT_FOUND"`;
